rand_distr/src/normal.rs

```diff
@@ -98,20 +98,25 @@ pub struct Normal {
     std_dev: f64,
 }
 
+/// Error type returned from `Normal::new` and `LogNormal::new`.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum Error {
+    /// `std_dev < 0`.
+    StdDevTooSmall,
+}
+
 impl Normal {
     /// Construct a new `Normal` distribution with the given mean and
     /// standard deviation.
-    ///
-    /// # Panics
-    ///
-    /// Panics if `std_dev < 0`.
     #[inline]
-    pub fn new(mean: f64, std_dev: f64) -> Normal {
-        assert!(std_dev >= 0.0, "Normal::new called with `std_dev` < 0");
-        Normal {
+    pub fn new(mean: f64, std_dev: f64) -> Result<Normal, Error> {
+        if std_dev < 0.0 {
+            return Err(Error::StdDevTooSmall);
```

Comment:
While standard deviations by definition are never negative, we can handle them easily. I suppose however that could prove surprising.

Reply:
We were panicking before, so I just stayed consistent with the previous behavior. Maybe open a new issue?